fish-rust/src/builtins/shared.rs

```diff
@@ -115,6 +123,22 @@ impl io_streams_t {
     pub fn ffi_ref(&self) -> &builtins_ffi::io_streams_t {
         unsafe { &*self.streams }
     }
+
+    pub fn stdin_is_directly_redirected(&self) -> bool {
+        self.ffi_ref().ffi_stdin_is_directly_redirected()
+    }
+
+    pub fn stdin_fd(&self) -> Option<BorrowedFd> {
```

Comment:
other ports use `RawFd`. I hope this is not causing needless friction.
If we change types like this we need to think ahead and have a clear idea of how to port all related code.
Looks like this implicitly converts to `i32` (I guess we should change the `read_blocked` parameter to `RawFd` at least)

Reply:
I've changed it to `RawFd`, and `read_blocked` now also takes a `RawFd`.